src/e3/anod/spec.py

```diff
@@ -387,6 +394,101 @@ def bind_to_sandbox(self, sandbox: SandBox) -> None:
             name=self.build_space_name, platform=self.env.platform
         )
 
+    def check_dll_closure(
+        self,
+        prefix: str | None = None,
+        ignored_libs: list[str] | None = None,
+        ldd_output: str | None = None,
+    ) -> None:
+        """Sanity check the shared library closure.
+
+        Make sure that the libraries only depend on authorized system shared
+        libraries.
+
+        Raise an error when a problem is detected.
+
+        It is possible to use the *ldd_output* parameter to mimic the ``ldd``
+        call. In that case, the output should look like::
+
+            /a/b/c.so:
+                    d.so => /e/f/d.so
+                    g.so => /a/b/g.so
+            /a/b/h.so:
+                    g.so => /a/b/g.so
+
+        :param prefix: The path where to find the library to be checked.
+        :param ignored_libs: The list of additional system libraries authorized
+            in the closure.
+        :param ldd_output: An ``ldd`` command output.
+
+        :raise: AnodError if some of the shared libraries in *prefix* (or in
+            *ldd_output*) is not in the same directory as the analysed element.
+        """
+        ignored: list[str] = ignored_libs or []
+        errors: dict[str, list[str]] = {}
+        lib_file: str = ""
+
+        if ldd_output is None:
+            # If ldd is not in the path, return and issue a warning.
+            if not which("ldd"):
+                self.log.warning("DLL closure check skipped: ldd could not be found.")
```

Comment:
Why not return an error in that case? The user might think everything is okay even though no check has been run.

Reply:
Because it could be run on Windows which may not have `ldd` in the path, and (I think) we do not want the build to fail in that case.

Reply:
I think we want the build to fail if we're in prod and we're relying on that feature to ensure that the libs are correct. We could raise a warning in USER mode if the call to that function raise an error but keep the error in PROD mode. This function should raise an error if ldd is not found.

Reply:
Done